apps/launch/components/Layout/SwitchNetworkSelect/useHandleRoute.tsx

```diff
@@ -0,0 +1,122 @@
+import { toLower } from 'lodash';
+import { useRouter } from 'next/router';
+import { useCallback, useEffect } from 'react';
+
+import {
+  ALL_SUPPORTED_CHAINS,
+  PAGES_TO_LOAD_WITHOUT_CHAINID,
+} from 'common-util/config/supportedChains';
+import { URL } from 'common-util/constants/urls';
+import { useGetChainIdFromPath } from 'common-util/hooks/useNetworkHelpers';
+import { useAppDispatch } from 'store/index';
+import { setNetworkId, setNetworkName } from 'store/network';
+
+const isValidNetworkName = (name?: string) => {
```

Comment:
seems that we have the same function in `apps/launch/common-util/hooks/useNetworkHelpers.tsx`